src/network/core/address.h

```diff
@@ -138,11 +138,15 @@ class NetworkAddress {
 	 */
 	int CompareTo(NetworkAddress &address)
 	{
+#ifndef __EMSCRIPTEN__
 		int r = this->GetAddressLength() - address.GetAddressLength();
 		if (r == 0) r = this->address.ss_family - address.address.ss_family;
 		if (r == 0) r = memcmp(&this->address, &address.address, this->address_length);
 		if (r == 0) r = this->GetPort() - address.GetPort();
 		return r;
+#else
+		return strcmp(GetHostname(), address.GetHostname());
```

Comment:
explanation please.

also it would be easier to read if the ifndef was inverted, imo